Cache errors in my docker container

I've been getting these cache errors for a while now
```
FAILED test_error_estimation.py::TestForm2Indicator::test_cell_integral - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/35/ae1780ef18766bc45...
FAILED test_error_estimation.py::TestGetDWRIndicator::test_convert_adjoint_error - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/35/ae1780ef18766bc45...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_time_partition_different_timesteps - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/c2/970828b5081f5dde3...
FAILED test_mesh_seq.py::TestExceptions::test_missing_field_error_0_get_function_spaces - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/97/9109d58f540a7d51e...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_time_partition_same_timestep - petsc4py.PETSc.Error: error code -1
FAILED test_mesh_seq.py::TestExceptions::test_return_dict_error_0_get_function_spaces - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/97/9109d58f540a7d51e...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_time_partition_wrong_field_error - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/97/9109d58f540a7d51e...
FAILED test_mesh_seq.py::TestExceptions::test_notimplemented_error_0_get_function_spaces - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/27/677287ea1b95bce99...
FAILED test_error_estimation.py::TestGetDWRIndicator::test_convert_neither - petsc4py.PETSc.Error: error code -1
FAILED test_mesh_seq.py::TestExceptions::test_return_dict_error_1_get_initial_condition - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/27/53a618dcd4b74eafd...
FAILED test_mesh_seq.py::TestExceptions::test_notimplemented_error_1_get_solver - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/97/9109d58f540a7d51e...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_unit_time_instant - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/35/ae1780ef18766bc45...
FAILED test_mesh_seq.py::TestExceptions::test_solver_generator_error - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/97/9109d58f540a7d51e...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_unit_time_instant_abs_0 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/5e/851014dbe61bd5ff5...
FAILED test_mesh_seq.py::TestExceptions::test_unexpected_field_error_0_get_function_spaces - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/c2/970828b5081f5dde3...
FAILED test_mesh_seq.py::TestExceptions::test_plot_dim_error_0 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/e3/9de24bc990009f819...
FAILED test_error_estimation.py::TestIndicators2Estimator::test_unit_time_instant_abs_1 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/b2/4b0f64a6e5cd786e0...
FAILED test_mesh_seq.py::TestExceptions::test_unexpected_field_error_1_get_initial_condition - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/27/677287ea1b95bce99...
FAILED test_mesh_seq.py::TestExceptions::test_plot_dim_error_1 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/01/de3fcdb90acc6195d...
FAILED test_metric.py::TestMetricNormalisation::test_consistency_06 - petsc4py.PETSc.Error: error code -1
FAILED test_metric.py::TestMetricNormalisation::test_consistency_07 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/62/fddcbc01ff5ee7509...
FAILED test_metric.py::TestMetricNormalisation::test_consistency_01 - FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/firedrake/.cache/tsfc/62/fddcbc01ff5ee7509...
```

and I thought it's just my docker image since we don't get these errors on the CI here. But now I pulled the new image and still get them. All other tests pass.

@jwallwork23 do you get these as well?

--- a/test/conftest.py
+++ b/test/conftest.py
@@ -15,15 +15,3 @@ def pytest_configure(config):
         "markers",
         "slow: mark test as slow to run",
     )
-
-
-def pytest_runtest_teardown(item, nextitem):
-    """
-    Clear caches after running a test
-    """
-    from firedrake.tsfc_interface import clear_cache
-    from pyop2.caching import clear_memory_cache
-    from pyop2.mpi import COMM_WORLD
-
-    clear_cache()
-    clear_memory_cache(COMM_WORLD)

--- a/test/conftest.py
+++ b/test/conftest.py
@@ -1,7 +1,7 @@
 """
 Global pytest configuration.
 
-**Disclaimer: some functions copied from firedrake/src/tests/conftest.py
+**Disclaimer: some functions copied from firedrake/tests/firedrake/conftest.py
 """
 
 
@@ -9,7 +9,7 @@ def pytest_configure(config):
     """
     Register an additional marker.
 
-    **Disclaimer: copied from firedrake/src/tests/conftest.py
+    **Disclaimer: copied from firedrake/tests/firedrake/conftest.py
     """
     config.addinivalue_line(
         "markers",

